projects/utils/src/lib/abstracts/query-param-form-sync/query-param-form-sync.component.abstract.ts

```diff
@@ -0,0 +1,168 @@
+import { Directive, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, Subject, filter, take, takeUntil, tap } from 'rxjs';
+
+type StringifiedQueryParamsType<QueryParamsType> = {
+	[key in keyof QueryParamsType]: string;
+};
+
+@Directive()
+export abstract class NgxQueryParamFormSyncComponent<
+	QueryParamsType,
+	FormType extends AbstractControl
+> implements OnInit, OnDestroy
+{
+	protected readonly destroyed$: Subject<void> = new Subject();
+
+	/**
+	 * The form in which we will save the queryParam data
+	 */
+	public form: FormType;
+
+	/**
+	 * The query params we wish to form
+	 */
+	protected queryParams$: Observable<StringifiedQueryParamsType<QueryParamsType>> = this.route
```

Comment:
Aangezien deze exposed wordt is het misschien ook interessant om de type te exporten voor mocht dat nodig zijn.

Reply:
Done!